import {Canvas, Meta, Story} from '@storybook/addon-docs';
import {withRouter} from 'storybook-addon-react-router-v6';

import Card from 'components/Card';
import {ConfigDecorator, LayoutDecorator} from 'story-utils/decorators';

import CancelAppointment from './CancelAppointment';
import {mockAppointmentCancelErrorPost, mockAppointmentCancelPost} from './mocks';

<Meta
  title="Private API / Appointments / Cancellation / Cancel"
  component={CancelAppointment}
  decorators={[LayoutDecorator, withRouter, ConfigDecorator]}
/>

## Cancel form with success handler

The error message after redirect can be ignored, it's because the full router is not wired up.

<Canvas>
  <Story
    name="Default"
    parameters={{
      reactRouter: {
        searchParams: {
          time: '2023-06-13T10:00:00+02:00',
          submission_uuid: '999c6769-bc2e-43d6-a2e0-5bf104f5130e',
        },
      },
      msw: {
        handlers: [mockAppointmentCancelPost],
      },
    }}
  >
    <CancelAppointment />
  </Story>
</Canvas>

## Missing time parameter

<Canvas>
  <Story name="Missing time parameter">
    <Card title="Error">
      <CancelAppointment />
    </Card>
  </Story>
</Canvas>

## Missing submission UUID parameter

<Canvas>
  <Story
    name="Missing submission parameter"
    parameters={{reactRouter: {searchParams: {time: '2023-06-13T10:00:00+02:00'}}}}
  >
    <Card title="Error">
      <CancelAppointment />
    </Card>
  </Story>
</Canvas>

## With backend validation errors

<Canvas>
  <Story
    name="Backend validation errors"
    parameters={{
      reactRouter: {
        searchParams: {
          time: '2023-06-13T10:00:00+02:00',
          submission_uuid: '999c6769-bc2e-43d6-a2e0-5bf104f5130e',
        },
      },
      msw: {
        handlers: [mockAppointmentCancelErrorPost],
      },
    }}
  >
    <CancelAppointment />
  </Story>
</Canvas>
